core/domain/interaction_jobs_one_off.py

```diff
@@ -392,3 +393,32 @@ def get_invalid_values(value_type, value, choices):
     @staticmethod
     def reduce(key, values):
         yield (key, values)
+
+
+class LogicProofInteractionOneOffJob(jobs.BaseMapReduceOneOffJobManager):
+    """Job that produces a list of (exploaration, state) pairs that use the
+    logicProof interaction. This will be used to send manual emails.
+    This job can be deleted after logicProof is removed from the codebase.
+    """
+
+    @classmethod
+    def entity_classes_to_map_over(cls):
+        return [exp_models.ExplorationModel]
+
+    @staticmethod
+    def map(item):
+        if item.deleted:
+            return
+        exploration = exp_fetchers.get_exploration_from_model(item)
+        for state in exploration.states.values():
+            if state.interaction.id == 'LogicProof':
+                owner_ids = rights_manager.get_exploration_rights(
+                    item.id).owner_ids
+                for user_id in owner_ids:
+                    user_mail = user_services.get_email_from_user_id(
+                        user_id)
+                    yield (item.id, 'user email is: %s' % (user_mail))
```

Comment:
Please yield 'SUCCESS' for every item that we handle.

Reply:
Also, can we yield `(email, item.id)` instead?

If we do it like you implemented it the output will be like this:
```
'expId1', ['user email is: mail@me.me', 'user email is: mail2@me.me']
'expId2', ['user email is: mail@me.me']
'expId3', ['user email is: mail3@me.me']
```
If we do `(email, item.id)` the output will be like this:
```
'mail@me.me', ['expId1', 'expId2']
'mail2@me.me', ['expId1']
'mail3@me.me', ['expId3']
```

Reply:
Done